src/pages/blog/{prismicBlogPost.uid}.tsx

```diff
@@ -0,0 +1,120 @@
+import { graphql, PageProps } from "gatsby";
+import React from "react";
+import BlogPostTemplate from "./BlogPostTemplate";
+
+/**
+  This is the setup for our dynamically generated blog post pages.
+  All urls of the pattern "/blog/{uid}"" point here and perform the query on this page.
+  Refer to the following docs:
+  https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/
+  https://prismic.io/docs/technologies/tutorial-4-add-dynamic-pages-gatsby
+*/
+
+export const query = graphql`
+  query PrismicBlogPost($uid: String!) {
+    prismicBlogPost(uid: { eq: $uid }) {
+      data {
+        author {
+          text
+        }
+        header_image {
+          alt
+          copyright
+          url
+        }
+        title {
+          text
+        }
+        publish_date
+        body {
+          ... on PrismicBlogPostDataBodyStatsBlock {
+            id
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodyImageWithCaption {
+            primary {
+              image {
+                url
+                alt
+              }
+              caption {
+                raw
+              }
+            }
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodyCtaBlock {
+            id
+            primary {
+              button_link {
+                url
+              }
+              button_text {
+                text
+              }
+              header {
+                text
+              }
+            }
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodySeparator {
+            id
+            primary {
+              separator_image {
+                url
+              }
+            }
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodyQuoteBlock {
+            id
+            primary {
+              quote {
+                text
+              }
+            }
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodyTextBlock {
+            id
+            primary {
+              body_text {
+                html
+              }
+            }
+            slice_type
+          }
+          ... on PrismicBlogPostDataBodyFileDownloadBlock {
+            id
+            primary {
+              file {
+                url
+              }
+              file_download_header {
+                text
+              }
+            }
+            slice_type
+          }
+        }
+      }
+    }
+  }
+`;
+
+const PrismicBlogPostPage = ({
+  data,
+}: PageProps<GatsbyTypes.PrismicBlogPostQuery>) => {
+  if (!data?.prismicBlogPost?.data) return <h1>There was a problem</h1>;
+  const blogData = data.prismicBlogPost.data;
+
+  return (
+    <BlogPostTemplate
+      title={blogData?.title?.text || ""}
+      author={blogData?.author?.text || ""}
```

Comment:
```suggestion
      title={blogData?.title?.text ?? ""}
      author={blogData?.author?.text ?? ""}
```

There's a better operator to use here, the [`??` nullish coalescing operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing), which only uses the value on the right-hand side if the left-hand side is null or undefined.

Reply:
I keep forgetting about that, thanks! I'll swap to using `??` here in my next branch, since I'll be touching this component a lot anyway